src/plugins/external_tool_support/src/blast_plus/AlignToReferenceBlastDialog.cpp

```diff
@@ -214,23 +218,33 @@ QList<Task *> AlignToReferenceBlastCmdlineTask::onSubTaskFinished(Task *subTask)
         config.arguments << argString.arg(RESULT_ALIGNMENT_ARG).arg(QFileInfo(settings.resultAlignmentFile).absoluteFilePath());
 
         config.reportFile = reportFile.fileName();
+        config.errorStateFile = errorStateFile.fileName();
         config.emptyOutputPossible = true;
 
         config.logLevel = LogLevel_TRACE;
 
         cmdlineTask = new CmdlineInOutTaskRunner(config);
         result.append(cmdlineTask);
-    } else if (subTask == cmdlineTask && settings.addResultToProject) {
-        // add load document task
-        FormatDetectionConfig config;
-        QList<FormatDetectionResult> formats = DocumentUtils::detectFormat(settings.resultAlignmentFile, config);
-        CHECK_EXT(!formats.isEmpty() && (NULL != formats.first().format), setError(tr("wrong output format")), result);
+    } else if (subTask == cmdlineTask) {
+        bool opened = errorStateFile.open();
+        SAFE_POINT_EXT(opened, setError(L10N::errorOpeningFileWrite(errorStateFile.fileName())), result);
+
+        QString subprocessError = errorStateFile.readAll();
+        errorStateFile.close();
+        CHECK_EXT(subprocessError.isEmpty(), setError(subprocessError), result);
 
-        DocumentFormat *format = formats.first().format;
-        CHECK_EXT(format->getSupportedObjectTypes().contains(GObjectTypes::MULTIPLE_CHROMATOGRAM_ALIGNMENT), setError(tr("wrong output format")), result);
+        if (settings.addResultToProject) {
+            // add load document task
+            FormatDetectionConfig config;
+            QList<FormatDetectionResult> formats = DocumentUtils::detectFormat(settings.resultAlignmentFile, config);
+            CHECK_EXT(!formats.isEmpty() && (NULL != formats.first().format), setError(tr("wrong output format")), result);
```

Comment:
detectFormat must never return nullptrs in the list: the check is not needed.

Reply:
This is not my code, but OK, removed